### PR TITLE
Enable AO WCDA test

### PR DESCRIPTION
**Pull request by aerorahul**
# Description
This PR:
- adds GSI + SOCA C48 5-deg ocean 3DVar test (courtesy @guillaumevernieres)
- adds a toggle to optionally disable `ocnanalvrfy` job.  It is possible this is taking a while for the first time as it is likely downloading cartopy shapefiles.  @CoryMartin-NOAA reminded me that I had put a PR in cartopy that will allow users to use downloaded shapefiles from a shared space.  Perhaps we can place those in the glopara fix space and load the address to it via a modulefile.  Something TBD in the future. 

@aerorahul:
- ran the first half-cycle of this test to ensure all the ICs are staged and the updates to `setup_expt.py` are made to enable a `warm_start` with `IAU` for the atmosphere.  This is a 3DVar test for the atmosphere and hence there is a single atmosphere increment.  Future development will incorporate 4D increments if desired.
- facilitate the staging of the fix data and possibly the observations `DMPDIR` for oceanobs processing.  This is not the usual `DMPDIR`, it is only ocean obs, used in the `prepoceanobs` step.
- also ran the full test on Hera.  It can be viewed at `/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNTESTS/EXPDIR/socat1`

@guillaumevernieres will:
- finalize `obs_list.yaml` and other updates needed in the GDASApp.  @guillaumevernieres please feel free to list what those updates are as relevant (or not!)

This test will only be introduced on Hera; Orion will follow soon after.

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO

# How has this been tested?
Testing on Hera is ongoing

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by aerorahul** [comment]
Temporarily disabled all other CI tests to allow testing just the SOCA test

**Comment by aerorahul** [comment]
@TerrenceMcGuinness-NOAA 
You may need to update your copy of CI scripts on Hera.
I had to enable the checkout of GDASapp in order to run the test added in this PR.
Please let me know how you wish to proceed.

**Comment by TerrenceMcGuinness-NOAA** [comment]
Ok @aerorahul added the -u switch to the `checkout.sh`  in the CI dev cron scripts on Hera and Orion and restated this PR test on Hera.

**Comment by aerorahul** [comment]
@guillaumevernieres 
The test failed.
When you get a chance, would you be able to help us understand why?
`/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1963/RUNTESTS/COMROT/C48_atm3DVar_ocnhybDA_0a6d7218/logs/2021032318/gdasocnanalrun.log`
Is it because `obs/` is empty and `observations` in `var.yaml` is too?
Seems likely that the experiment is not being configured correctly.

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres The test failed. When you get a chance, would you be able to help us understand why? `/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1963/RUNTESTS/COMROT/C48_atm3DVar_ocnhybDA_0a6d7218/logs/2021032318/gdasocnanalrun.log` Is it because `obs/` is empty and `observations` in `var.yaml` is too? Seems likely that the experiment is not being configured correctly.

Will do @aerorahul . Where's the rundir?

**Comment by guillaumevernieres** [comment]
@aerorahul , the issue is the content of the `obs_list.yaml`. The list of obs in that file matches what we use in the gdasapp ci but none of these obs are in the r2d2 db that you point to.
 
I'll issue a PR in the gdasapp later today.

In the mean time, if you want to test the rest of the system, here's an `obs_list.yaml` that should work:
```yaml
observers:
- !INC ${OBS_YAML_DIR}/adt_all.yaml
- !INC ${OBS_YAML_DIR}/adt_j3_egm2008.yaml
- !INC ${OBS_YAML_DIR}/adt_3a_egm2008.yaml
- !INC ${OBS_YAML_DIR}/adt_3b_egm2008.yaml
- !INC ${OBS_YAML_DIR}/adt_c2_egm2008.yaml
- !INC ${OBS_YAML_DIR}/adt_sa_egm2008.yaml
- !INC ${OBS_YAML_DIR}/sst_viirs_npp_l3u_so025.yaml
- !INC ${OBS_YAML_DIR}/sst_metopa_l3u_so025.yaml
- !INC ${OBS_YAML_DIR}/sst_metopb_l3u_so025.yaml
- !INC ${OBS_YAML_DIR}/sst_metopc_l3u_so025.yaml
- !INC ${OBS_YAML_DIR}/icec_ssmis_f17_north.yaml
- !INC ${OBS_YAML_DIR}/icec_ssmis_f17_south.yaml
- !INC ${OBS_YAML_DIR}/icec_ssmis_f18_north.yaml
- !INC ${OBS_YAML_DIR}/icec_ssmis_f18_south.yaml
```

**Comment by aerorahul** [comment]
> @aerorahul , the issue is the content of the `obs_list.yaml`. The list of obs in that file matches what we use in the gdasapp ci but none of these obs are in the r2d2 db that you point to.
> 
> I'll issue a PR in the gdasapp later today.
> 
> In the mean time, if you want to test the rest of the system, here's an `obs_list.yaml` that should work:
> 
> ```yaml
> observers:
> - !INC ${OBS_YAML_DIR}/adt_all.yaml
> - !INC ${OBS_YAML_DIR}/adt_j3_egm2008.yaml
> - !INC ${OBS_YAML_DIR}/adt_3a_egm2008.yaml
> - !INC ${OBS_YAML_DIR}/adt_3b_egm2008.yaml
> - !INC ${OBS_YAML_DIR}/adt_c2_egm2008.yaml
> - !INC ${OBS_YAML_DIR}/adt_sa_egm2008.yaml
> - !INC ${OBS_YAML_DIR}/sst_viirs_npp_l3u_so025.yaml
> - !INC ${OBS_YAML_DIR}/sst_metopa_l3u_so025.yaml
> - !INC ${OBS_YAML_DIR}/sst_metopb_l3u_so025.yaml
> - !INC ${OBS_YAML_DIR}/sst_metopc_l3u_so025.yaml
> - !INC ${OBS_YAML_DIR}/icec_ssmis_f17_north.yaml
> - !INC ${OBS_YAML_DIR}/icec_ssmis_f17_south.yaml
> - !INC ${OBS_YAML_DIR}/icec_ssmis_f18_north.yaml
> - !INC ${OBS_YAML_DIR}/icec_ssmis_f18_south.yaml
> ```

Thankyou @guillaumevernieres 
I will wait for an update to the GDASApp hash with the corrected OBS_LIST_YAML.
Please let me know when the GDASApp has been updated and a subsequent PR into the global-workflow updating the hash of GDASApp has been made.

**Comment by aerorahul** [comment]
Too many issues with running SOCA out of the box.  
Closing this PR and will disable testing of GDASapp.

**Comment by aerorahul** [comment]
@guillaumevernieres 
An experiment was setup with the hash [21c9a0a](https://github.com/NOAA-EMC/global-workflow/pull/1963/commits/21c9a0a62e8374520b3a155d333c29242f10cd42) to run the first half cycle with warm starts and with IAU
The details are:
```
EXPDIR: /scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNTESTS/EXPDIR/socatest
COMROOT: /scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNTESTS/COMROOT/socatest
```

The forecast for the first half-cycle was successful.

File: workflow/setup_expt.py [hunk 1]
```diff
@@ -219,11 +219,18 @@ def link_files_from_src_to_dst(src_dir, dst_dir):
...
     makedirs_if_missing(dst_dir)
-    for ftype in ['abias', 'abias_pc', 'abias_air', 'radstat']:
+    for ftype in ['abias', 'abias_pc', 'abias_air', 'radstat', 'atminc.nc']:
         fname = f'{inputs.cdump}.t{idatestr[8:]}z.{ftype}'
         src_file = os.path.join(src_dir, fname)
         if os.path.exists(src_file):
             os.symlink(src_file, os.path.join(dst_dir, fname))
+    # First 1/2 cycle also needs a atmos increment if doing warm start
+    if inputs.start in ['warm']:
+        for ftype in ['atmi003.nc', 'atminc.nc', 'atmi009.nc']:
```
Comment (guillaumevernieres) [inline workflow/setup_expt.py:229]: That section fails if the increment files are already linked @aerorahul

Comment (aerorahul) [reply workflow/setup_expt.py:229]: good catch.  removed the one on 222

**Comment by aerorahul** [comment]
Kick off CI on Hera.
ICs have been staged in glopara space for this test.  Obs still need to be placed. 
🤞🏾 hoping for a successful test

**Comment by guillaumevernieres** [comment]
Where is the output of the CI @aerorahul ? I'd like to check before we declare victory.

**Comment by aerorahul** [comment]
> Where is the output of the CI @aerorahul ? I'd like to check before we declare victory.

The test removes the `COM` and `EXPDIR` if the CI is successful to recover diskspace.

File: ci/cases/yamls/soca_gfs_defaults_ci.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+defaults:
+  !INC {{ HOMEgfs }}/parm/config/gfs/yaml/defaults.yaml
+base:
+  ACCOUNT: {{ 'SLURM_ACCOUNT' | getenv }}
+  DO_JEDIOCNVAR: "NO"
```
Comment (guillaumevernieres) [inline ci/cases/yamls/soca_gfs_defaults_ci.yaml]: Is that `DO_JEDIOCNVAR` switch flipped back on somewhere else? If not then we probably didn't run `soca` in the test.

Comment (aerorahul) [reply ci/cases/yamls/soca_gfs_defaults_ci.yaml]: oops

**Comment by AndrewEichmann-NOAA** [comment]
@aerorahul, @guillaumevernieres is having me take care of the `DO_JEDIOCNVAR` switch - where do you think it should be acted on?

**Comment by guillaumevernieres** [comment]
> @aerorahul, @guillaumevernieres is having me take care of the `DO_JEDIOCNVAR` switch - where do you think it should be acted on?

Not really @AndrewEichmann-NOAA , we talked about a switch to toggle the marine DA verify task. `DO_JEDIOCNVAR` already exist. The option should go in `config.base`.

**Comment by AndrewEichmann-NOAA** [comment]
> > @aerorahul, @guillaumevernieres is having me take care of the `DO_JEDIOCNVAR` switch - where do you think it should be acted on?
> 
> Not really @AndrewEichmann-NOAA , we talked about a switch to toggle the marine DA verify task. `DO_JEDIOCNVAR` already exist. The option should go in `config.base`.

@guillaumevernieres Sorry, I misunderstood the purpose of `DO_JEDIOCNVAR`.  I'm asking about where in the scripts the option (`DO_VERFOCNDA`?) should be acted on - the rocoto script, the jjob, etc.

**Comment by aerorahul** [comment]
@guillaumevernieres @AndrewEichmann-NOAA 
I have made the `ocnanalvrfy` task optional in [58ef58d](https://github.com/NOAA-EMC/global-workflow/pull/1963/commits/58ef58d28827c9a94d9762c44292aef92fe0b5da)

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres @AndrewEichmann-NOAA I have made the `ocnanalvrfy` task optional in [58ef58d](https://github.com/NOAA-EMC/global-workflow/pull/1963/commits/58ef58d28827c9a94d9762c44292aef92fe0b5da)

thanks @aerorahul

**Comment by aerorahul** [comment]
@guillaumevernieres 
The obs have been moved to glopara space.
There are a couple of TODO's waiting for your input, if you would please comment, that would be great.  We won't hold this PR up to resolve those, but your feedback will be greatly appreciated.
I kicked off the CI with the updated obs address and when it passes, this PR is ready to merge. 🤞🏾

**Review by guillaumevernieres — APPROVED** [review]
Thanks @aerorahul !

**Review by guillaumevernieres — APPROVED** [review]
Thanks @aerorahul !

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**